src/solve-quadratic/solveQuadraticEquation.js

```diff
@@ -46,20 +46,23 @@ export default function solveQuadraticEquation(a, b, c) {
 
     // Calculate real part (common for all cases)
     const realPart = negB.div(twoA);
+    // Convert -0 to +0 if needed
+    const realValue = realPart.toNumber();
+    const normalizedReal = realValue === 0 ? 0 : realValue;
 
     if (delta.gt(0)) {
         const sqrtDelta = delta.sqrt().div(twoA);
         return [
-            { real: realPart.plus(sqrtDelta).toNumber(), imaginary: 0 },
-            { real: realPart.minus(sqrtDelta).toNumber(), imaginary: 0 }
+            { real: normalizedReal + sqrtDelta.toNumber(), imaginary: 0 },
```

Comment:
We should not have imaginary = 0 or real = 0 by default!
If there is no root, better to leave it as undefined or null.